### PR TITLE
Note that insecure serving is deprecated

### PR DESCRIPTION
**Pull request by tengqm**

### CODE REVIEW

**Comment by steveperry-53** [comment]
/approve (for writing)
Still need `/lgtm` for technical accuracy.

**Comment by steveperry-53** [comment]
/cc @lavalamp 
/cc @erictune
/cc @bowei 
Could one of you review this for technical accuracy? Thanks.

**Comment by liggitt** [comment]
this PR is not accurate. insecure ports were marked deprecated in 1.10, but are still present and active by default. see comment at https://github.com/kubernetes/kubernetes/pull/59018/files#r177849954 which notes that marking them deprecated hides the presence of the flag and the related help, which is not good

**Comment by tengqm** [comment]
@liggitt Will revise the PR to note that the feature is "deprecated".

File: docs/admin/authorization/rbac.md [hunk 1]
```diff
@@ -693,11 +693,12 @@ subjects:
   name: user-1
 ```
 
-When bootstrapping the first roles and role bindings, it is necessary for the initial user to grant permissions they do not yet have.
-To bootstrap initial roles and role bindings:
-
-* Use a credential with the `system:masters` group, which is bound to the `cluster-admin` super-user role by the default bindings.
-* If your API server runs with the insecure port enabled (`--insecure-port`), you can also make API calls via that port, which does not enforce authentication or authorization.
```
[the remark this answers is not in the document]
Comment (tengqm) [reply docs/admin/authorization/rbac.md]: done.

**Comment by Bradamant3** [comment]
@liggitt can you please look again? I'm still concerned about removing options. And if insecure ports are still active, should we continue to include some more of the old information about localhost? But if it's ok, if you /lgtm we can get this PR merged. Thanks!

**Comment by liggitt** [comment]
> @liggitt can you please look again? I'm still concerned about removing options. And if insecure ports are still active, should we continue to include some more of the old information about localhost?

I agree. It is fine to add notes that the insecure option is deprecated but I would not remove it or refer to it as if it is not a current behavior until it is actually gone.

**Comment by chenopis** [comment]
@liggitt So, what's the status of this PR. Does it still require more changes, or is it good to go?

**Comment by Bradamant3** [comment]
@tengqm it looks to me as though Jordan wants more of the insecure option content reinstated, per response to my question ^^ (and I agree with him, bc deprecated doesn't mean gone and especially wrt localhost this seems like important and useful information). lmk if you disagree. In the meantime, I'll see whether I can find out what's happening to insecure ports for 1.11.

**Comment by tengqm** [comment]
The fact is ... there are many "hidden" options in v1.10 for kubelet and kube-apiserver. These options are not visible when you do 'kubelet --help', they are not shown in the generated reference docs. The current reference doc generator even doesn't have a way to detect the existence of such options. If we want to continue document them, we should be consistent across all options. If we decide to omit them be cause they are effectively "gone", we again should be consistent.

**Comment by liggitt** [comment]
That is considered a bug that has been corrected in master and should be corrected in the 1.10 branch if possible

**Comment by Bradamant3** [comment]
Note also that it's not just deprecated options that don't show up in the generated docs. the `enable-admission-plugins` flag is also missing, and my guess is that other new options are in similar state. (I've filed an issue.) If options are hidden in generated docs, all the more reason to keep the content wherever we can do so manually until we get the generated docs right.

(This also seems like a larger doc and code and possibly governance issue. Adding an item to the sig-docs agenda and will try to find a better forum for cross-sig discussion too.)

**Comment by zacharysarah** [comment]
@tengqm 👋 We've migrated the site's framework to Hugo. Please re-open this PR with changes to files in their post-migration location (`/content/en/docs/...`) and reference this PR in the new PR's description.

**Pull request closed without merging**